go-controller/pkg/ovn/namespace.go

```diff
@@ -376,20 +385,41 @@ func (oc *Controller) getNamespaceLocked(ns string) *namespaceInfo {
 	return nsInfo
 }
 
-// createNamespaceLocked locks namespacesMutex, creates an entry for ns, and returns it
+// ensureNamespaceLocked locks namespacesMutex, gets/creates an entry for ns, and returns it
 // with its mutex locked.
-func (oc *Controller) createNamespaceLocked(ns string) *namespaceInfo {
+func (oc *Controller) ensureNamespaceLocked(ns string) *namespaceInfo {
 	oc.namespacesMutex.Lock()
-	defer oc.namespacesMutex.Unlock()
-
-	nsInfo := &namespaceInfo{
-		networkPolicies:       make(map[string]*networkPolicy),
-		podExternalRoutes:     make(map[string]map[string]string),
-		multicastEnabled:      false,
-		routingExternalPodGWs: make(map[string]gatewayInfo),
+	nsInfo := oc.namespaces[ns]
+	nsInfoExisted := false
+	if nsInfo == nil {
+		nsInfo = &namespaceInfo{
+			networkPolicies:       make(map[string]*networkPolicy),
+			podExternalRoutes:     make(map[string]map[string]string),
+			multicastEnabled:      false,
+			routingExternalPodGWs: make(map[string]gatewayInfo),
+		}
+		// we are creating nsInfo and going to set it in namespaces map
+		// so safe to hold the lock while we create and add it
+		defer oc.namespacesMutex.Unlock()
+		oc.namespaces[ns] = nsInfo
+	} else {
+		nsInfoExisted = true
+		// if we found and existing nsInfo, do not hold the namespaces lock
```

Comment:
nit: and -> an

Reply:
if there are more things to fix in this PR i'll fix that, but if its just a couple of nits I'll defer to a later PR